rust/src/api/wallet.rs

```diff
@@ -443,4 +441,40 @@ mod tests {
 
     //     // TODO: more cases
     // }
+
+    #[test]
+    fn test_unblinded_detection() {
+        let network = Network::Testnet;
+        let mnemonic = "";
+        let descriptor = Descriptor::new_confidential(network, mnemonic.to_string())
+            .expect("Failed to create descriptor");
+        let wallet = Wallet::init(network, "/tmp/lwk_test".to_string(), descriptor).expect("");
+
+        let confidential_address = wallet.address_last_unused().unwrap();
+        println!("The confidential address is {}", confidential_address);
+
+        //Write code to halt for the payment to be made. maybe we can add a timer? or take an input from user that confirms that he did the pyment or not.
+
+        let electrum_url = "les.bullbitcoin.com:995".to_string();
+        wallet.sync(electrum_url.clone()).unwrap();
+
+        let balance_before = wallet.balances().unwrap();
+        let txs_before = wallet.txs().unwrap();
+
+        assert!(balance.get(0), >0);
+        assert!(!txs_before.is_empty());
+
+        let blinding_key = wallet.blinding_key().unwrap();
+        let unblinded_address = "";
+        let unblinded_address = println!("The unblinded address is {}", unblinded_address);
+        println!("Fund the unblinded address");
```

Comment:
I'll have to create a function that converts `unblinded_address` to `blinded` using the binding key.
I couldn't find any method that does this either in `lwk-dart` repo or in `lwk`.

Reply:
you mean blinded to unblinded? We don't need to go the other way around.

```
let unblinded_address = confidential_address.standard;
```

Reply:
This struct has methods to go either direction:

https://docs.rs/elements/0.24.1/elements/address/struct.Address.html